tests/test_odp_event_manager.py

```diff
@@ -0,0 +1,273 @@
+            "data": {"key-1": "value1"}
+        },
+        {
+            "type": "t2",
+            "action": "a2",
+            "identifiers": {"id-key-2": "id-value-2"},
+            "data": {"key-2": "value2"}
+        }
+    ]
+
+    processed_events = [
+        {
+            "type": "t1",
+            "action": "a1",
+            "identifiers": {"id-key-1": "id-value-1"},
+            "data": {
+                "idempotence_id": test_uuid,
+                "data_source_type": "sdk",
+                "data_source": "python-sdk",
+                "data_source_version": __version__,
+                "key-1": "value1"
+            }
+        },
+        {
+            "type": "t2",
+            "action": "a2",
+            "identifiers": {"id-key-2": "id-value-2"},
+            "data": {
+                "idempotence_id": test_uuid,
+                "data_source_type": "sdk",
+                "data_source": "python-sdk",
+                "data_source_version": __version__,
+                "key-2": "value2"
+            }
+        }
+    ]
+
+    def test_odp_event_init(self):
+        with mock.patch('uuid.uuid4', return_value=self.test_uuid):
+            event = OdpEvent(**self.events[0])
+        self.assertEqual(event, self.processed_events[0])
+
+    def test_odp_event_manager_success(self):
+        mock_logger = mock.Mock()
+        event_manager = OdpEventManager(self.odp_config, mock_logger)
+        event_manager.start()
+
+        with mock.patch('requests.post', return_value=self.fake_server_response(status_code=200)):
+            event_manager.send_event(**self.events[0])
+            event_manager.send_event(**self.events[1])
+            event_manager.stop()
+
+        self.assertEqual(len(event_manager._current_batch), 0)
+        mock_logger.error.assert_not_called()
+        mock_logger.debug.assert_any_call('Flushing batch size 2.')
+        mock_logger.debug.assert_any_call('Received ODP event shutdown signal.')
+        self.assertStrictFalse(event_manager.is_running)
+
+    def test_odp_event_manager_batch(self):
+        mock_logger = mock.Mock()
+        event_manager = OdpEventManager(self.odp_config, mock_logger)
+        event_manager.start()
+
+        event_manager.batch_size = 2
+        with mock.patch.object(
+            event_manager.zaius_manager, 'send_odp_events', return_value=False
+        ) as mock_send, mock.patch('uuid.uuid4', return_value=self.test_uuid):
+            event_manager.send_event(**self.events[0])
+            event_manager.send_event(**self.events[1])
+            event_manager.event_queue.join()
+
+        mock_send.assert_called_once_with(self.api_key, self.api_host, self.processed_events)
+        self.assertEqual(len(event_manager._current_batch), 0)
+        mock_logger.error.assert_not_called()
+        mock_logger.debug.assert_any_call('Flushing ODP events on batch size.')
+        event_manager.stop()
+
+    def test_odp_event_manager_multiple_batches(self):
```

Comment:
I like this test on multiple batches, and the one after, for multiple flushes. There are always going to be many batches flushed in real scenarios.